connexion/operations/openapi.py

```diff
@@ -0,0 +1,376 @@
+        to any shadowed built-ins
+        :type pythonic_params: bool
+        :param uri_parser_class: class to use for uri parseing
+        :type uri_parser_class: AbstractURIParser
+        """
+        self.components = components or {}
+
+        def component_get(oas3_name):
+            return self.components.get(oas3_name, {})
+
+        # operation overrides globals
+        security_schemes = component_get('securitySchemes')
+        app_security = operation.get('security', app_security)
+        uri_parser_class = uri_parser_class or OpenAPIURIParser
+
+        super(OpenAPIOperation, self).__init__(
+            api=api,
+            method=method,
+            path=path,
+            operation=operation,
+            resolver=resolver,
+            app_security=app_security,
+            security_schemes=security_schemes,
+            validate_responses=validate_responses,
+            strict_validation=strict_validation,
+            randomize_endpoint=randomize_endpoint,
+            validator_map=validator_map,
+            pythonic_params=pythonic_params,
+            uri_parser_class=uri_parser_class
+        )
+
+        self._definitions_map = {
+            'components': {
+                'schemas': component_get('schemas'),
+                'requestBodies': component_get('requestBodies'),
+                'parameters': component_get('parameters'),
+                'securitySchemes': component_get('securitySchemes'),
+                'responses': component_get('responses'),
+                'headers': component_get('headers'),
+            }
+        }
+
+        # todo support definition references
+        # todo support references to application level parameters
+        self._request_body = operation.get('requestBody')
+        if self._request_body:
+            self._request_body = self._resolve_reference(self._request_body)
+
+        def resolve_parameters(parameters):
+            return [self._resolve_reference(p) for p in parameters]
+
+        self.parameters = resolve_parameters(operation.get('parameters', []))
+        if path_parameters:
+            self.parameters += resolve_parameters(path_parameters)
+
+        def resolve_responses(responses):
+            if not responses:
+                return responses
+            responses = deepcopy(responses)
+            for status_code, resp in responses.items():
+                # check components/responses
+                if '$ref' in resp:
+                    ref = self._resolve_reference(resp)
+                    del resp['$ref']
+                    resp = ref
+
+                content = resp.get("content", {})
+                for mimetype, resp in content.items():
+                    # check components/examples
+                    examples = resp.get("examples", {})
+                    for _, example in examples.items():
+                        example = self._resolve_reference(example)
```

Comment:
What is the intention here? the `examples` variable seems to be overridden every time with no use, even right after the loop on line 132.